include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -596,17 +607,20 @@ __pattern_transform_scan_single_group(_ExecutionPolicy&& __exec, _InRng&& __in_r
             const bool __is_full_group = __n == __wg_size;
 
             if (__is_full_group)
-                return __parallel_transform_scan_single_group_submitter<
+                return __parallel_transform_scan_static_single_group_submitter<
```

Comment:
May we combine this and next call of ```__parallel_transform_scan_static_single_group_submitter``` into one call ?
```/* _IsFullGroup= */ true```, ```/* _IsFullGroup= */ false``` -> ```__is_full_group```;

Reply:
The `__is_full_group` variable is a runtime value. We have to pass a different compile-time boolean to the class based on the runtime boolean.

Reply:
May something be like ```__is_full_group ? ::std::true_type{} : ::std::false_type{}``` ?

Reply:
It doesn't work just like that, because we cannot (easily) jump from run-time at compile-time and then back. It's an template argument

Reply:
Both branches of a ternary expression must be implicitly convertible to the same type. 

We can do something cleaner by making a function that dispatches to a function with a static boolean template parameter based on a runtime boolean value. Something like:
```c++
template<typename _F>
auto __dispatch_boolean(const bool __b, _F&& __f)
{
    if (__b)
        std::forward<_F>(__f).template operator()<true>();
    else
        std::forward<_F>(__f).template operator()<false>();
}
```
We could then wrap the call to `__parallel_transform_scan_static_single_group_submitter` in a functor whose `operator()` is templated with a boolean. It could be cleaner, but I'm not sure if that this PR is the best place to do that.

Reply:
From my point of view - up to you.